### ISSUE TITLE
Graph not showing (empty box instead)

### ISSUE THREAD

**Issue by szawadski** [xavividal/kanboard-plugin-relationgraph#12]
Hello,

Tried your nice plugin that gives all its value to the "internal link" feature in Kanboard.
Unfortunately, no graph is showing when I launch the relation graph page related to 1 task.
I am using:
- Kanboard 1.0.32
- PHP 5.6
- Kanboard is installed on the webserver of my Synology DS412+ with DSM 6.0
  Tried with different types of internal links but same problem.

Any idea of what I did wrong? Thanks in advance for your help.

Here are a few screenshots.
![image](https://cloud.githubusercontent.com/assets/18078729/17694732/90e3665c-63a5-11e6-9ba7-667841cdf342.png)
![image](https://cloud.githubusercontent.com/assets/18078729/17694749/a9edfa4a-63a5-11e6-9138-dd1d5421547a.png)
![image](https://cloud.githubusercontent.com/assets/18078729/17694782/cec8bf30-63a5-11e6-8588-665a96f8a342.png)

Regards,

Sébastien

**Comment by oliviermaridat** [comment]
Any news on this issue?

**Comment by xavividal** [comment]
Sorry, I was on holidays and now I'm back. 

It seems a JS error. Can you take a look to the JS console to see if there are some kind of warning or error?

**Comment by oliviermaridat** [comment]
Yes, this is a JS error, a CSP one. But I don't understand why... Are your script loading something from the Web outside of Kanboard?
I know there is a new Plugin method to override CSP rules: https://github.com/kanboard/kanboard/blob/master/doc/plugin-overrides.markdown#override-http-content-security-policy

JS Console outputs:

> Content Security Policy: the parameters of the page prevent loading of a resource at http://media/data/Web/Test/kanboard/plugins/Relationgraph/Asset/Javascript/vis/vis.js?1472643914 (« default-src http://kanboard.local »).
> Content Security Policy: the parameters of the page prevent loading of a resource at http://media/data/Web/Test/kanboard/plugins/Relationgraph/Asset/Javascript/vis/vis.css?1472643914 (« style-src http://kanboard.local 'unsafe-inline' »).
> Content Security Policy: the parameters of the page prevent loading of a resource at http://media/data/Web/Test/kanboard/plugins/Relationgraph/Asset/Javascript/GraphBuilder.js?1472643914 (« default-src http://kanboard.local »).

**Comment by szawadski** [comment]
Hello!
Also some news on my side. Installed the plug-in on the same server but another instance of Kanboard (prod) and... it works perfectly. Nevertheless, I do not understand why it does not want to work on my test database (same config/version as the prod, except the path is /kanboard-test/ instead of /kanboard/)

Regards,

Sébastien

**Comment by Boobert** [comment]
I'm having the same issue of an empty box showing up.

My Chrome dev console reads:

GET http://localhost/kanboard/C:/Apache24/htdocs/kanboard/plugins/Relationgraph/Asset/Javascript/vis/vis.js?1469542501 
localhost/:316 GET http://localhost/kanboard/C:/Apache24/htdocs/kanboard/plugins/Relationgraph/Asset/Javascript/vis/vis.css?1469542501 
localhost/:316 GET http://localhost/kanboard/C:/Apache24/htdocs/kanboard/plugins/Relationgraph/Asset/Javascript/GraphBuilder.js?1469542501 403 (Forbidden)

Am trying to track down where those funky urls are being generated.

**Comment by seccon06** [comment]
Solved like this ....
Same problem here - the plugin worked on one installation I had done some weeks ago on another domain. Same versions, same server. I noticed the permissions where different, but amending them was not working.
So, I downloaded the working Relationship plugin and uploaded it to the installation that had the blank window.
All working now.

**Comment by ghost** [comment]
We have the same problem. Was working fine until we upgraded to 1.0.35
PHP: 5.4.16, Kanboard: 1.0.35, Centos 7

Chrome console reports: Refused to load the script 'http://var/www/html/kanboard/plugins/Relationgraph/Asset/Javascript/vis/vis.js?1479719313' because it violates the following Content Security Policy directive: "default-src 'self'". Note that 'script-src' was not explicitly set, so 'default-src' is used as a fallback.

**Comment by lvic** [comment]
Hello

I have a similiar problem like Boobert here (on a local installation via xampp):
Empty box instead of a graph, chrome console log shows that GET requests like http://localhost/core/kanboard/C:/xampp/htdocs/core/kanboard/plugins/Relationgraph/Asset/Javascript/vis/vis.js?1479719313  are returning errors as the path to assets obviously isn't valid.

Rendered HTML from kanboard looks like this:
![image](https://cloud.githubusercontent.com/assets/9221213/21439494/b1fbca5c-c88e-11e6-8696-3ca2c6cc2a35.png)

Related PHP section should be this:
![image](https://cloud.githubusercontent.com/assets/9221213/21439548/fefc7fa4-c88e-11e6-9af4-3e37d3e2cca0.png)

Somehow the path doesn't get resolved properly. 
Is this js function in php above from the plugin or part of kanboard? 
And which component provides this PLUGINS_DIR variable?


__EDIT:__
By hooking onto asset hooks in plugin initialization I was able to get it running:
![image](https://cloud.githubusercontent.com/assets/9221213/21441491/7cadcada-c898-11e6-8951-e97724f151e5.png)

Now I see the graph and can interact within. Awesome, thank you!

**Comment by ghost** [comment]
@lvic Thanks! After these two changes the plugin works.
